api/apps/api/src/modules/projects/projects.controller.ts

```diff
@@ -323,13 +324,19 @@ export class ProjectsController {
   @Get(':id/calibration')
   async getProjectBlmValues(
     @Param('id') id: string,
+    @Req() req: RequestWithAuthenticatedUser,
   ): Promise<ProjectBlmValuesResponseDto> {
-    const result = await this.projectsService.findProjectBlm(id);
+    const result = await this.projectsService.findProjectBlm(id, req.user.id);
 
     if (isLeft(result))
-      throw new NotFoundException(
-        `Could not find project BLM values for project with ID: ${id}`,
-      );
+      switch (result.left) {
+        case forbidden:
+          throw new ForbiddenException();
+        default:
+          throw new NotFoundException(
+            `Could not find project BLM values for project with ID: ${id}`,
+          );
```

Comment:
without exhaustive check it can be overlooked and throw 404 for something totally unrelated - actually any error, so can be misleading.

```
const error = response.left;
    switch (error) {
      case metadataNotFound:
      case outputMetadataNotFound:
        throw new NotFoundException(`Marxan was not yet executed.`);
      case outputZipNotYetAvailable:
      case marxanFailed:
      case inputZipNotYetAvailable:
        throw new NotFoundException(
          `Marxan has not yet finished or finished with error.`,
        );
      default:
        const _exhaustiveCheck: never = error;
        throw _exhaustiveCheck;
    }
```
(this is, if any new option is added, TS/IDE will force you to add there something)

Reply:
I just added the same thing to handle ACL errors properly to this PR and apart from that improved how the errors are handled here I think so it could only throw an 404 for a `projectNotFound` error. All others will remain the same though.

Reply:
@rubvalave still missing `_exhaustiveCheck` to ensure we can react once some new error is added to possible outcomes.

Reply:
I think it is sorted out now